src/app-layer-ssl.c

```diff
@@ -2764,9 +2764,8 @@ static int SSLStateGetEventInfoById(int event_id, const char **event_name,
 
 static int SSLRegisterPatternsForProtocolDetection(void)
 {
-    if (AppLayerProtoDetectPMRegisterPatternCS(IPPROTO_TCP, ALPROTO_TLS,
-                                               "|01 00 02|", 5, 2, STREAM_TOSERVER) < 0)
-    {
+    if (AppLayerProtoDetectPMRegisterPatternCSwPP(IPPROTO_TCP, ALPROTO_TLS, "|01 00 02|", 5, 2,
+                STREAM_TOSERVER, SSLProbingParser, 5, 5) < 0) {
```

Comment:
Both `pp_min_depth` and `pp_max_depth` are `5` ... seems odd that `min == max`

Reply:
You are right, other `SSLProbingParser` use 0 and 3

In this case, that does not make much difference since we have the pattern which needs at least 5 bytes